Updates to Nested Queries Documentation (#26640)
DataDog/documentation: content/en/metrics/nested_queries.md

{{< img src="/metrics/nested_queries/nested-queries-higher-res-json.png" alt="example of higher resolution queries using nested queries in the JSON" style="width:100%;" >}}
{{% /collapse-content %}} 

## Moving rollup

> Datadog provides a `moving_rollup` function that enables aggregation of data points over a specified time window (quick refresher [here][10]). By using nested queries, you can extend its functionality to incorporate lookback mode, allowing you to analyze data points beyond the original query window. This provides a more comprehensive view of your query's trends and patterns over the specified time window.

Reviewer: ```suggestion
Datadog provides a `moving_rollup` function that enables aggregation of datapoints over a specified time window. See [moving-rollup][10] for more information. By using nested queries, you can extend its functionality to incorporate lookback mode, allowing you to analyze datapoints beyond the original query window. This provides a more comprehensive view of your query's trends and patterns over the specified time window.
```

{{< img src="/metrics/nested_queries/moving-rollup-diagram.png" alt="example of old vs. new moving_rollup function" style="width:100%;" >}}

The existing version of the `moving-rollup` function only supports the following aggregators:
- `avg`
- `sum`
- `min`
- `max`
- `median`

When nesting queries, only the lookback mode version of the `moving_rollup` function can be used. This version of the function supports the following aggregators:
- `avg`
- `sum`
- `min`
- `max`
- `count`
- `count by`
- `arbitrary percentile pxx` (`p78, p99, p99.99, etc.`)
- `stddev`

{{% collapse-content title="Max Moving rollup with Lookback Mode Enabled" level="h5" %}}
When nesting these `moving_rollups`, the rollup intervals provided must get larger as shown in the UI or JSON tab:

{{< img src="/metrics/nested_queries/moving_rollup1_ui.png" alt="example of moving rollup in the UI" style="width:100%;" >}}

{{< img src="/metrics/nested_queries/moving_rollup1_json.png" alt="example of moving rollup in the JSON" style="width:100%;" >}}

{{% /collapse-content %}} 

{{% collapse-content title="Standard Deviation Moving Rollup with Lookback Mode Enabled" level="h5" %}}

> You can also use percentiles and standard deviation with the new moving rollup function that supports lookback and nest moving rollups that have lookback enabled as well. 

Reviewer: ```suggestion 
You can also use percentiles and standard deviation with the new moving rollup function, which supports lookback, and allows nesting of moving rollups with lookback enabled.
```

In the UI or JSON tab, it would look as follows:

{{< img src="/metrics/nested_queries/moving_rollup2_ui.png" alt="example of moving rollup with standard deviation in the UI" style="width:100%;" >}}

{{< img src="/metrics/nested_queries/moving_rollup2_json.png" alt="example of moving rollup with standard deviation in the JSON" style="width:100%;" >}}

{{% /collapse-content %}} 

## Boolean threshold remapping functions

> Remap functions allow you to refine and transform query results based on specific conditions, extending functionality for monitoring and analysis. Nested queries unlocks the following 3 new functions:

Reviewer: ```suggestion
Remap functions allow you to refine and transform query results based on specific conditions, extending functionality for monitoring and analysis. Nested queries unlocks the following three new functions:
```
